src/scripts/main.js

```diff
@@ -1,3 +1,428 @@
 'use strict';
 
-// write code here
+const types = {
+  number: 'number',
+  text: 'text',
+  salary: 'salary',
+};
+
+const table = document.querySelector('table');
+const thead = document.querySelector('thead');
+const tbody = document.querySelector('tbody');
+
+const tableRows = table.rows;
+
+const headRow = tableRows[0];
+const headElements = [...headRow.children];
+
+const newEmployee = {
+  name: '',
+  position: '',
+  office: '',
+  age: 0,
+  salary: 0,
+};
+
+const dataType = (value) => {
+  switch (value) {
+    case 'Age':
+      return types.number;
+
+    case 'Salary':
+      return types.salary;
+
+    case 'Name':
+      return types.text;
+
+    case 'Position':
+      return types.text;
+
+    case 'Office':
+      return types.text;
+
+    default:
+      return types.text;
+  }
+};
+
+// #region sort
+const ascOrder = 'asc';
+const descOrder = 'desc';
+
+let sortTitle = '';
+let order = ascOrder;
+
+thead.addEventListener('click', (e) => {
+  const appendTable = (list) => {
+    list.forEach((tr) => {
+      tbody.appendChild(tr);
+    });
+  };
+
+  const changeOrder = () => {
+    order = order === ascOrder ? descOrder : ascOrder;
+  };
+
+  const link = e.target.closest('th');
+
+  if (link.textContent === sortTitle) {
+    changeOrder();
+  } else {
+    order = ascOrder;
+  }
+
+  sortTitle = link.textContent;
+
+  if (!link) {
+    return;
+  }
+
+  const index = headElements.findIndex(
+    (el) => el.textContent === link.textContent,
+  );
+
+  const tbodyCopy = tbody.cloneNode(true);
+
+  const sortedList = [...tbodyCopy.children].sort((rowA, rowB) => {
+    const firstEl = Array.from(rowA.children)[index].textContent;
+    const secondEl = Array.from(rowB.children)[index].textContent;
+
+    if (dataType(link.textContent) === types.salary) {
+      const salaryNum = (salary) => Number(salary.slice(1).split(',').join(''));
+
+      return salaryNum(firstEl) - salaryNum(secondEl);
+    }
+
+    if (dataType(link.textContent) === types.number) {
+      return Number(firstEl) - Number(secondEl);
+    }
+
+    if (dataType(link.textContent) === types.text) {
+      return firstEl.localeCompare(secondEl);
+    }
+  });
+
+  tbody.textContent = '';
+
+  return order === ascOrder
+    ? appendTable(sortedList)
+    : appendTable(sortedList.reverse());
```

Comment:
The sorting logic does not take into account the current `order` when comparing values. Ensure that the comparison logic reflects the desired sorting order (ascending or descending) to avoid incorrect sorting results.